File: selenium4java/ExampleTest1.java
package newpackage;

import java.time.Duration;

import org.junit.After;
import org.junit.Before;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Wait;
import org.openqa.selenium.support.ui.WebDriverWait;

public class ExampleTest1 {

	WebDriver driver;
	
	@BeforeEach
	public void setUp() {
		// Instantiate ChromeDriver as Chrome will be used here
		driver = new ChromeDriver();
	}
	
	@Test
	void testExample1() throws InterruptedException {		
		// Goto the web site we want
		driver.navigate().to("https://restaurantinfo.com/");
		
		// Verify we are at restaurantinfo.com by its Logo image
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(3));
		wait.until(ExpectedConditions.visibilityOf(
			driver.findElement(By.cssSelector(".bg-fixed.bg-top img[src = 'https://restaurantinfo.com/images/logo/logo-desktop.png']"))
		));
		
		String searchTerm = "Chef";
		driver.findElement(By.cssSelector(".ais-Autocomplete input")).sendKeys(searchTerm);
		
		By firstMatchingSuggestionLocator = By.cssSelector(
			"*[id = autosuggest] ul > li .py-2 .text-xl.mx-4.cursor-pointer");
		
		// Verify autocomplete is displayed
		wait.until(ExpectedConditions.textToBe(firstMatchingSuggestionLocator, searchTerm));
		
		// Click on first matching suggestion
		driver.findElement(firstMatchingSuggestionLocator).click();
		
		driver.findElement(By.cssSelector("*[id = autosuggest] > input[name = 'city_state']")).sendKeys("New York");
		wait.until(ExpectedConditions.textToBePresentInElementLocated(
			By.cssSelector(".autosuggest__results-item span.text-xl"),
			"New York, NY, USA"
		));
		
		// Click on entry New York, NY, USA
		driver.findElement(firstMatchingSuggestionLocator).click();
		
		// Search
		driver.findElement(By.xpath(".//button//*[text() = 'Search']")).click();

		// Verify result page is shown with 'New York, NY, USA' under Location
		wait.until(ExpectedConditions.presenceOfElementLocated(
			By.cssSelector(".ais-RefinementList input[value = 'New York, NY, USA']")
		));
	}
	
	@AfterEach
	public void tearDown() {
		if (driver != null) {
			driver.close();
			driver.quit();
		}
	}
	
}

File: seleniumpython/ExampleTest1.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Chrome()
timeout = 10 

# Goto the web site we want
driver.get("https://restaurantinfo.com")

try:
  WebDriverWait(driver, timeout).until(
    EC.presence_of_element_located((By.CSS_SELECTOR, ".bg-fixed.bg-top img[src = 'https://restaurantinfo.com/images/logo/logo-desktop.png']"))
  )
except Exception as e:
  print("Cannot find 'logo-desktop.png'!!!")
  driver.quit()

searchTerm = "Chef"
driver.find_element(By.CSS_SELECTOR, ".ais-Autocomplete input").send_keys(searchTerm)

firstMatchingSuggestionCSS = "*[id = autosuggest] ul > li .py-2 .text-xl.mx-4.cursor-pointer"

# Verify autocomplete is displayed
try:
  WebDriverWait(driver, timeout).until(
    EC.text_to_be_present_in_element((By.CSS_SELECTOR, firstMatchingSuggestionCSS), searchTerm)
  )
except Exception as e:
  print("Cannot find 'Chef' in autocomplete!!!")
  driver.quit()

# Click on first matching suggestion
driver.find_element(By.CSS_SELECTOR, firstMatchingSuggestionCSS).click()

driver.find_element(By.CSS_SELECTOR, "*[id = autosuggest] > input[name = 'city_state']").send_keys("New York")
try:
  WebDriverWait(driver, timeout).until(
    EC.text_to_be_present_in_element((By.CSS_SELECTOR, ".autosuggest__results-item span.text-xl"), "New York, NY, USA")
  )
except Exception as e:
  print("Cannot find 'New York, NY, USA' in autocomplete!!!")
  driver.quit()

# Click on entry New York, NY, USA
driver.find_element(By.CSS_SELECTOR, firstMatchingSuggestionCSS).click();

# Search
driver.find_element(By.XPATH, ".//button//*[text() = 'Search']").click();

# Verify result page is shown with 'New York, NY, USA' under Location
try:
  WebDriverWait(driver, timeout).until(
    EC.presence_of_element_located((By.CSS_SELECTOR, ".ais-RefinementList input[value = 'New York, NY, USA']"))
  )
except Exception as e:
  print("Cannot find 'New York, NY, USA' under Location on result page!!!")
  driver.quit()


driver.quit()

